nucypher/blockchain/eth/agents.py

```diff
@@ -190,68 +150,84 @@ def owner(self):
 
 class NucypherTokenAgent(EthereumContractAgent):
 
-    registry_contract_name = NUCYPHER_TOKEN_CONTRACT_NAME
+    contract_name: str = NUCYPHER_TOKEN_CONTRACT_NAME
 
     @contract_api(CONTRACT_CALL)
-    def get_balance(self, address: str = None) -> int:
+    def get_balance(self, address: Optional[ChecksumAddress] = None) -> NuNits:
         """Get the NU balance (in NuNits) of a token holder address, or of this contract address"""
         address = address if address is not None else self.contract_address
         return self.contract.functions.balanceOf(address).call()
 
     @contract_api(CONTRACT_CALL)
-    def get_allowance(self, owner: str, spender: str) -> int:
+    def get_allowance(self, owner: ChecksumAddress, spender: ChecksumAddress) -> NuNits:
         """Check the amount of tokens that an owner allowed to a spender"""
-        return self.contract.functions.allowance(owner, spender).call()
+        allowance: int = self.contract.functions.allowance(owner, spender).call()
+        return NuNits(allowance)
 
     @contract_api(TRANSACTION)
-    def increase_allowance(self, sender_address: str, target_address: str, increase: int) -> dict:
-        contract_function = self.contract.functions.increaseAllowance(target_address, increase)
-        receipt = self.blockchain.send_transaction(contract_function=contract_function, sender_address=sender_address)
+    def increase_allowance(self,
+                           sender_address: ChecksumAddress,
+                           target_address: ChecksumAddress,
+                           increase: NuNits
+                           ) -> TxReceipt:
+        contract_function: ContractFunction = self.contract.functions.increaseAllowance(target_address, increase)
+        receipt: TxReceipt = self.blockchain.send_transaction(contract_function=contract_function, sender_address=sender_address)
         return receipt
 
     @contract_api(TRANSACTION)
-    def approve_transfer(self, amount: int, target_address: str, sender_address: str) -> dict:
+    def approve_transfer(self, amount: NuNits,
+                         target_address: ChecksumAddress,
+                         sender_address: ChecksumAddress
+                         ) -> TxReceipt:
         """Approve the transfer of tokens from the sender address to the target address."""
-        payload = {'gas': 500_000}  # TODO #842: gas needed for use with geth! <<<< Is this still open?
-        contract_function = self.contract.functions.approve(target_address, amount)
-        receipt = self.blockchain.send_transaction(contract_function=contract_function,
-                                                   payload=payload,
-                                                   sender_address=sender_address)
+        payload: TxParams = {'gas': Wei(500_000)}  # TODO #842: gas needed for use with geth! <<<< Is this still open?
```

Comment:
I guess this should not have wei, because it's not gas price, it's gas limit

Reply:
Following the annotations here: <https://github.com/ethereum/web3.py/blob/master/web3/types.py#L167>

Reply:
ok, it confuses, but what we can do

Reply:
:heavy_check_mark: